http2/transport.go

```diff
@@ -634,10 +644,10 @@ func (t *Transport) newClientConn(c net.Conn, singleUse bool) (*ClientConn, erro
 		tconn:                 c,
 		readerDone:            make(chan struct{}),
 		nextStreamID:          1,
-		maxFrameSize:          16 << 10,           // spec default
-		initialWindowSize:     65535,              // spec default
-		maxConcurrentStreams:  1000,               // "infinite", per spec. 1000 seems good enough.
-		peerMaxHeaderListSize: 0xffffffffffffffff, // "infinite", per spec. Use 2^64-1 instead.
+		maxFrameSize:          16 << 10,                    // spec default
+		initialWindowSize:     65535,                       // spec default
+		maxConcurrentStreams:  initialMaxConcurrentStreams, // "infinite", per spec. Use a smaller value until we have received server settings.
```

Comment:
Are there any restrictions that preventing setting `maxConcurrentStreams` in `Transport` configurations?

Reply:
That could be done but it only has meaning for http2

Reply:
> it only has meaning for http2

What do you mean?

Reply:
Transport configurations come from: https://pkg.go.dev/net/http#Transport which is used for http1 and http2, just flagging that adding an option for initial max concurrent streams would be http2 specific.

Reply:
By using https://pkg.go.dev/net/http#Transport with function https://pkg.go.dev/golang.org/x/net/http2#ConfigureTransports limited the connection reuse cases:
https://github.com/golang/net/blob/ad29c8ab022f0f2c6a18c8279b1db2230acfa59b/http2/transport.go#L173-L176
This may result in an **http2: no cached connection was available** error from the lines below:
https://github.com/golang/net/blob/ad29c8ab022f0f2c6a18c8279b1db2230acfa59b/http2/transport.go#L476-L480

Aren't you modifying the http2 package source code? Why shouldn't it have a `MaxConcurrentStreams` specific configuration anyway?